pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/groupby/DefaultGroupByExecutor.java

```diff
@@ -81,10 +83,11 @@ public DefaultGroupByExecutor(QueryContext queryContext, ExpressionContext[] gro
     // Initialize group key generator
     int numGroupsLimit = queryContext.getNumGroupsLimit();
     int maxInitialResultHolderCapacity = queryContext.getMaxInitialResultHolderCapacity();
-    if (hasNoDictionaryGroupByExpression) {
+    if (hasNoDictionaryGroupByExpression || _nullHandlingEnabled) {
       if (groupByExpressions.length == 1) {
         _groupKeyGenerator =
-            new NoDictionarySingleColumnGroupKeyGenerator(transformOperator, groupByExpressions[0], numGroupsLimit);
+            new NoDictionarySingleColumnGroupKeyGenerator(transformOperator, groupByExpressions[0], numGroupsLimit,
+                _nullHandlingEnabled);
       } else {
         _groupKeyGenerator =
```

Comment:
Add a TODO to support MV and dictionary based. No dictionary performance is much worse than dictionary based